analyses/infercnv-consensus-cell-type/notebooks/01_ewings-consensus-cell-types.Rmd

```diff
@@ -0,0 +1,186 @@
+
+```{r}
+immune_celltypes <- readr::read_tsv(immune_url) |>
+  dplyr::pull(consensus_annotation)
+
+celltype_df <- celltype_files |>
+  purrr::map(readr::read_tsv) |>
+  purrr::list_rbind() |>
+  dplyr::select(library_id, barcodes, ewing_annotation, consensus_annotation) |>
+  dplyr::mutate(
+    category = dplyr::case_when(
+      consensus_annotation == "Unknown" ~ "Unknown",
+      consensus_annotation %in% immune_celltypes ~ "immune",
+      .default = "other"
+    )
+  )
+```
+
+
+```{r}
+category_count_df <- celltype_df |>
+  dplyr::count(library_id, category) |>
+  tidyr::pivot_wider(
+    names_from = category,
+    values_from = n,
+    values_fill = 0
+  )
+
+# this way we can print the full table at once; RMarkdown will only show 10 rows at once
+knitr::kable(category_count_df)
+```
+It seems most immune cells are concentrated in 2-3 samples, and most samples don't have any immune cells.
+
+How many immune cells are there total?
+
+```{r}
+total_immune <- sum(category_count_df$immune)
+total_immune
+```
+
+`inferCNV` authors [recommend that at least 100 cells](https://github.com/broadinstitute/infercnv/issues/188) be used for a "normal" reference, so we are above that threshold.
+
+If we were to use all of these, what fraction is that per sample?
+We'll just divide the total number of cells by this count, not taking into consideration how many immune cells are actually in that sample, to get a rough sense of how many immune cells there are relative to all.
+
+```{r}
+celltype_df |>
+  dplyr::count(library_id, name = "total_cells") |>
+  dplyr::mutate(fraction_immune = round(total_immune / total_cells, 2)) |>
+  knitr::kable()
+```
+
+For the low-quality sample `SCPCL001111` this fraction isn't very meaningful since there are fewer cells in the sample than there are pooled immune cells.
+For all other samples, the fraction ranges from 0.03 - 0.13.
+
+
+Let's look a bit more at the breakdown of immune cell types:
+
+```{r}
+celltype_df |>
+  dplyr::filter(category == "immune") |>
+  dplyr::count(library_id, consensus_annotation) |>
+  # this will arrange columns (after pivoting) in descending order of total counts
+  dplyr::arrange(desc(n)) |>
+  tidyr::pivot_wider(
+    names_from = consensus_annotation,
+    values_from = n,
+    values_fill = 0
+  )
+```
+
+
+Let's look at the total number of macrophages and all types of T cell, since these are the dominant cell types here.
+
+```{r}
+celltype_df |>
+  dplyr::mutate(
+    consensus_annotation = ifelse(
+      stringr::str_detect(consensus_annotation, "T cell"),
+      "T cell type",
+      consensus_annotation
+    )
+  ) |>
+  dplyr::filter(consensus_annotation %in% c("T cell type", "macrophage")) |>
+  dplyr::count(consensus_annotation)
+```
+These constitute the vast majority of cells, so it would also be an option to use just these groups of cell types.
+If we were to want a single immune cell type, then macrophage is the only suitable option to get more than 100 cells.
```

Comment:
I don't think we want to use a single immune cell type, but choosing the most dominant immune cell types and comparing that to using all immune cells makes sense to me.